x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/badge/span_links_badge.tsx

```diff
@@ -47,7 +55,14 @@ export function SpanLinksBadge({ linkedParents, linkedChildren, id }: Props) {
         </EuiFlexGroup>
       }
     >
-      <EuiBadge data-test-subj={`spanLinksBadge_${id}`}>
+      <EuiBadge
+        data-test-subj={`spanLinksBadge_${id}`}
+        onClick={(e: any) => {
+          e.stopPropagation();
```

Comment:
Tangential to this:

Currently the full "waterfall row" is clickable which will open the transaction/span details flyout.
We now have a few clickable elements on the waterfall (span links, related errors etc) and we need to work around the fact that the full row is clickable by adding `e.stopPropagation`. As a user, it is also confusing what exactly causes the flyout to open, and what doesn't. Copying text is also no possible. 

Suggestion: the row itself should not be clickable. Instead we have a dedicated icon where the flyout can be opened. This is similar to the agent health management screen. cc @boriskirov

Reply:
We don't know if that is confusing for the user, but I agree we can make it way clearer and polish a lot of the interaction around the waterfall. I think we can create a pitch around fixing the waterfall and list all those items there and others collected by @cauemarcondes. Thanks @sqren